pkg/reconcile/pipeline/handler/project/impl.go

```diff
@@ -160,6 +165,10 @@ func BindingsAsFiles(ctx pipeline.Context) {
 	if !ctx.BindAsFiles() {
 		return
 	}
+	if len(ctx.BindingItems()) == 0 {
+		stop(ctx, errors.New("no binding item to project"))
+		return
+	}
```

Comment:
see above comment on `PreFlightCheck` - once there, this check can be removed.

Reply:
+1